### PR TITLE
Fix function synchronization

### PR DESCRIPTION
**Pull request by m00g3n**
**Description**

Changes proposed in this pull request:
- Fix function synchronization for functions that were not created using function API

**Related issue(s)**

See also: https://github.com/kyma-incubator/hydroform/issues/137

### CODE REVIEW

File: function/pkg/operator/triggers_operator.go [hunk 1]
```diff
@@ -4,22 +4,50 @@ import (
...
+	"github.com/kyma-incubator/hydroform/function/pkg/resources/types"
 	"github.com/pkg/errors"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
 )
 
 const message = "ownerID"
 
+type FnRef struct {
```
Comment (tgorgol) [inline function/pkg/operator/triggers_operator.go:16]: Maybe we could make this type private, I don't think it's used anywhere outside this package?

File: function/pkg/operator/triggers_operator.go [hunk 2]
```diff
@@ -4,22 +4,50 @@ import (
...
 		Client: c,
 		items:  u,
+		fnRef: FnRef{
+			name:      fnName,
+			namespace: fnNamespace,
+		},
+	}
+}
+
+func buildPredicate(fnRef FnRef, items []unstructured.Unstructured) func(map[string]interface{}) (bool, error) {
```
Comment (tgorgol) [inline function/pkg/operator/triggers_operator.go:38]: I think it's worth to add some short comment describing how this predicate behaves.

File: function/pkg/operator/triggers_operator.go [hunk 3]
```diff
@@ -30,7 +58,10 @@ func (t triggersOperator) Apply(ctx context.Context, opts ApplyOptions) error {
 	if !found {
```
Comment (tgorgol) [inline function/pkg/operator/triggers_operator.go:58]: Do we still need the `ownerID` label? Couldn't it be removed now that we're no longer using it for deletion?

File: function/pkg/operator/triggers_operator.go [hunk 4]
```diff
@@ -4,42 +4,62 @@ import (
...
 		items:  u,
+		fnRef: functionReference{
+			name:      fnName,
+			namespace: fnNamespace,
+		},
+	}
+}
+
+// buildMattchRemovedTriggerPredicate - creates a predicate to match the triggers that should be deleted
+func buildMattchRemovedTriggerPredicate(fnRef functionReference, items []unstructured.Unstructured) func(map[string]interface{}) (bool, error) {
```
Comment (tgorgol) [inline function/pkg/operator/triggers_operator.go:36]: There's a small typo in "buildMat**t**chRemovedTriggerPredicate".

**Review by tgorgol — APPROVED** [review]
LGTM

**Review by pPrecel — APPROVED** [review]

**Pull request merged**